### DISCUSSION TITLE
Why is Advanced Data Protection not supported?

### DISCUSSION THREAD

**Discussion by neilschark** [icloud-photos-downloader/icloud_photos_downloader#960] [Ideas]
Unfortunately, ADP is not supported. The docs say 
```
ADP
Advanced Data Protection (ADP) for iCloud accounts is not supported because icloudpd simulates web access, which is disabled with ADP.
```

But the stated reason is not true. You can use iCloud-Web with ADP, your Keys get on the fly send from one of your devices to the servers to encrypt the data once. 

It may be a problem for autoamted syncing the data every night or something, but for manually triggering a download/sync it should not be a problem.

**Comment by AndreyNikiforov** [comment]
I don't have ADP enabled and haven't experimented with it. From what I am reading about it, web access can be enabled for short period of time (1hr?). It is unclear if that can be enabled from `icloudpd` code or has to be enabled through icloud.com. The limitation of 1hr makes this feature not of high interest and complication with implementation reduces value further. I suspect that a good research/analysis/experimentation may change the balance.

**Reply by binaryanomaly** [reply]
Any updates / progress here?
icloudp looks exactly like what I would need/want but not supporting ADP is kinda a deal breaker.
